patches/tModLoader/Terraria.ModLoader/NPCLoader.cs

```diff
@@ -946,6 +946,84 @@ private class HookList
 			}
 		}
 
+		private static HookList HookCanGoToKingStatue = AddHook<Func<NPC, bool?>>(g => g.CanGoToKingStatue);
+
+		public static bool CanGoToKingStatue(NPC npc) {
+			bool defaultCanGo = npc.modNPC?.CanGoToKingStatue() ?? false;
+
+			switch (npc.type) {
+				case NPCID.Merchant:
+				case NPCID.Guide:
+				case NPCID.ArmsDealer:
+				case NPCID.Demolitionist:
+				case NPCID.Clothier:
+				case NPCID.GoblinTinkerer:
+				case NPCID.Wizard:
+				case NPCID.SantaClaus:
+				case NPCID.Truffle:
+				case NPCID.DyeTrader:
+				case NPCID.Painter:
+				case NPCID.WitchDoctor:
+				case NPCID.Pirate:
+				case NPCID.LightningBug: // A literal bug
+				case NPCID.Angler:
+				case NPCID.DD2Bartender:
+					defaultCanGo = true;
+					break;
+			}
```

Comment:
as much as I like this, couldn't you get the default value from the vanilla statement? rather than replacing it just assign it to a flag, that would make it more update compatible